olive/cache.py

```diff
@@ -9,82 +9,103 @@
 from pathlib import Path
 from typing import Dict, Optional, Union
 
-from olive.common.config_utils import serialize_to_json
-from olive.common.utils import hash_dict
-from olive.resource_path import ResourcePath, create_resource_path
+from olive.common.config_utils import ConfigBase, convert_configs_to_dicts, serialize_to_json, validate_config
+from olive.common.utils import hash_dict, set_nested_dict_value
+from olive.resource_path import ResourcePath, create_resource_path, find_all_resources
 
 logger = logging.getLogger(__name__)
 
 
-def get_cache_sub_dirs(cache_dir: Union[str, Path] = ".olive-cache"):
-    """Return the subdirectories of the cache directory.
+def set_cache_dir(cache_dir: Union[str, Path]):
+    """Set the cache directory."""
+    os.environ["OLIVE_CACHE_DIR"] = str(cache_dir)
 
-    There are three subdirectories: models, runs, and evaluations.
+
+def unset_cache_dir():
+    """Unset the cache directory."""
+    if "OLIVE_CACHE_DIR" in os.environ:
+        del os.environ["OLIVE_CACHE_DIR"]
+
+
+def get_cache_dir() -> Path:
+    """Return the cache directory.
+
+    The cache directory is set by the OLIVE_CACHE_DIR environment variable.
+    If the environment variable is not set, the default cache directory is .olive-cache.
     """
-    cache_dir = Path(cache_dir)
-    return cache_dir / "models", cache_dir / "runs", cache_dir / "evaluations", cache_dir / "non_local_resources"
+    return Path(os.environ.get("OLIVE_CACHE_DIR", ".olive-cache")).resolve()
 
 
-def clean_cache(cache_dir: Union[str, Path] = ".olive-cache"):
+def get_cache_sub_dirs() -> Dict[str, Path]:
+    """Return the subdirectories of the cache directory."""
+    cache_dir = get_cache_dir()
+    return {
```

Comment:
Do think if it is needed to return a class but not a dict? Then any down-stream users can directly use `dot` to get the models/evaluation/etc..

Reply:
this is a good idea. I can create a cache class with these subdirectories. And also put the cache utils as methods for it.

Reply:
for the time being I will just make the cache dirs as a class with dot notation. We can refactor all of the cache methods into a single class as a follow up since that requires more work that is out of scope for this PR.

Reply:
Moved all cache utils into an `OliveCache` class